packages/near-membrane-base/src/membrane.ts

```diff
@@ -217,6 +225,26 @@ export function init(
         };
     }
 
+    // This wrapping mechanism provides the means to add instrumentation
+    // to the callable functions used to coordinate work between the sides
+    // of the membrane.
+    // TODO: do we need to pass more info into instrumentation hooks?
+    function instrumentCallableWrapper<T extends (...args: any[]) => any>(fn: T, label: string): T {
+        if (instrumentation) {
+            return <T>function instrumentedFn(...args: any[]): any {
+                let result;
+                instrumentation.start(label);
+                try {
+                    result = fn(...args);
+                } finally {
+                    instrumentation.end(label);
+                }
+                return result;
+            };
+        }
+        return fn;
```

Comment:
this is the most important line of the PR, returning the provided fn (meaning noop) when instrumentation API is not provided.